services/venice-controller/src/main/java/com/linkedin/venice/controller/server/ControllerRoutes.java

```diff
@@ -64,13 +67,7 @@ public Route getLeaderController(Admin admin) {
       try {
         AdminSparkServer.validateParams(request, LEADER_CONTROLLER.getParams(), admin);
         String cluster = request.queryParams(CLUSTER);
-        responseObject.setCluster(cluster);
-        Instance leaderController = admin.getLeaderController(cluster);
-        responseObject.setUrl(leaderController.getUrl(isSslEnabled()));
-        if (leaderController.getPort() != leaderController.getSslPort()) {
-          // Controller is SSL Enabled
-          responseObject.setSecureUrl(leaderController.getUrl(true));
-        }
+        requestHandler.getLeaderController(new ControllerRequest(cluster), responseObject);
```

Comment:
So this function will populate the `responseObject`? Can we not call it a getter if that is the case? e.g., `populateLeaderController`, perhaps?